android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDataJsonUtil.java

```diff
@@ -158,9 +158,10 @@ public static void processKeyboardPackageUpdateJSON(Context aContext, JSONObject
                 // Cloud catalog has newer KMP version available
                 String updateKMP = kbd.getUpdateKMP();
                 if (updateKMP != null) {
-                  if (!updateKMP.equalsIgnoreCase(cloudKMP)) {
-                    // Update keyboard info with the latest KMP link
-                    kbd.setUpdateKMP(cloudKMP);
+                  if (!updateKMP.startsWith(cloudKMP)) {
```

Comment:
I am not entirely sure what the content of `updateKMP` and `cloudKMP` should be. But I can see two potential issues:

1. `startsWith` is case sensitive, so that's different to previously.
2. If I am understanding this correctly, this will now match `updateKMP == "my_keyboard_ex"` for `cloudKMP == "my_keyboard"`.

Reply:
> I am not entirely sure what the content of updateKMP and cloudKMP should be

They contain the "kmp" property returned from the api.keyman.com [package-version](https://api.keyman.com/package-version?platform=android&keyboard=sil_cameroon_qwerty) query. updateKMP is empty if there's no updates to make.

I changed the comparison to `startsWith()` to handle the scenario where previous checks had stored the appended languageID:
updateKMP = `https://keyman.com/go/package/download/keyboard/sil_cameroon_qwerty?version=6.0.8&update=1&bcp47=aku` and
cloudKMP = `https://keyman.com/go/package/download/keyboard/sil_cameroon_qwerty?version=6.0.8&update=1`

Reply:
I'm not very comfortable with making these assumptions on the structure of the URL. I think we should use a URL parser and do a more precise match rather than assume that there isn't already a bcp47 code in it, etc.

Reply:
I've refactored the URI validation to CloudDataJSONUtils where it checks if the cloud KMP link is a newer version.